components/FlexibleHeader/src/MDCFlexibleHeaderView.m

```diff
@@ -546,7 +546,11 @@ - (id)topSafeAreaGuide {
 #pragma mark - Private (fhv_ prefix)
 
 - (void)fhv_setContentOffset:(CGPoint)contentOffset {
-  _trackingScrollView.contentOffset = contentOffset;
+  // Avoid excessive writes. This can also cause infinite recursion if we're observing the content
+  // offset because of observesTrackingScrollViewScrollEvents.
+  if (!CGPointEqualToPoint(contentOffset, _trackingScrollView.contentOffset)) {
```

Comment:
huh, I always assumed UIKit did that internally.

Reply:
I think KVO hookery happens surrounding the property implementation, so even if they did check, the KVO events would still fire.